tree/tree/inc/TTree.h

```diff
@@ -439,7 +445,7 @@ class TTree : public TNamed, public TAttLine, public TAttFill, public TAttMarker
    virtual Long64_t        GetSelectedRows() { return GetPlayer()->GetSelectedRows(); }
    virtual Int_t           GetTimerInterval() const { return fTimerInterval; }
            TBuffer*        GetTransientBuffer(Int_t size);
-   virtual Long64_t        GetTotBytes() const { return fTotBytes; }
+   virtual Long64_t        GetTotBytes() const { std::lock_guard<std::mutex> sentry(fCounterMutex); return fTotBytes; }
```

Comment:
The  use of a local variable depends on when compilers run destructors, before or after copying the value out to memory/register for returning from the function.